backend/ownerSizeTotalWorker.js

```diff
@@ -0,0 +1,270 @@
+'use strict'
+
+import sbp from '@sbp/sbp'
+import { appendToIndexFactory, removeFromIndexFactory, updateSize } from './database.js'
+import { readyQueueName } from './genericWorker.js'
+
+const TASK_TIME_INTERVAL = 30e3
+
+// This file defines two different methods for doing size computations:
+// `backend/server/computeSizeTaskDeltas` and `backend/server/computeSizeTask`.
+// The latter does a full computation by looking at all subresources and adding
+// them up while the former does a delta-based computation and is event-driven.
+// Both are used, but at different times.
+// The delta / event-driven version is likely to be the most performant while
+// the server is running, so this is used for updates.
+// However, the full computation is less likely to give incorrect results and
+// requires fewer pieces of data to be stored. This is what's used when the
+// server starts up.
+// Because the delta version could theoretically lead to incorrect results
+// (meaning it's not self-healing), we mighr want in the future to also randomly
+// do a full computation on some contracts.
+
+const updatedSizeList = new Set()
+const updatedSizeMap = new Map()
+
+// Super-fast 8-bit hash for base58 with low bucket size standard deviation
+//  (i.e., the 256 buckets get about the same number of elements)
+const fastBase58Hash = (cid: string) => {
+  const len = cid.length
+  const a = cid.codePointAt(len - 2) || 0
+  const b = cid.codePointAt(len - 1) || 0
+  return ((a * 19) + (b + 19)) & 0xFF
+}
```

Comment:
Let me know if it's fine to use this custom hash function (which I made specifically to be pretty fast) or if we should use something like CRC32 or CRC16, that could be slower but have a slightly more uniform distribution (and also works with any values, while this has function is made specifically for the base58 alphabet and an input of exactly two characters).

Reply:
This hash function seems perfect, you did a great job on it!